labelbox/data/serialization/ndjson/classification.py

```diff
@@ -0,0 +1,212 @@
+        return cls(answer=text.answer, schema_id=schema_id)
+
+
+class NDChecklistSubclass(NDFeature):
+    answer: List[NDFeature]
+
+    def to_common(self) -> Checklist:
+        return Checklist(answer=[
+            ClassificationAnswer(schema_id=answer.schema_id)
+            for answer in self.answer
+        ])
+
+    @classmethod
+    def from_common(cls, checklist: Checklist,
+                    schema_id: Cuid) -> "NDChecklistSubclass":
+        return cls(answer=[
+            NDFeature(schema_id=answer.schema_id) for answer in checklist.answer
+        ],
+                   schema_id=schema_id)
+
+
+class NDRadioSubclass(NDFeature):
+    answer: NDFeature
+
+    def to_common(self) -> Radio:
+        return Radio(answer=ClassificationAnswer(
+            schema_id=self.answer.schema_id))
+
+    @classmethod
+    def from_common(cls, radio: Radio, schema_id: Cuid) -> "NDRadioSubclass":
+        return cls(answer=NDFeature(schema_id=radio.answer.schema_id),
+                   schema_id=schema_id)
+
+
+### ====== End of subclasses
+
+
+class NDText(NDAnnotation, NDTextSubclass):
+
+    @classmethod
+    def from_common(cls, text: Text, schema_id: Cuid, extra: Dict[str, Any],
+                    data: Union[TextData, RasterData]) -> "NDText":
+        return cls(
+            answer=text.answer,
+            dataRow={'id': data.uid},
+            schema_id=schema_id,
+            uuid=extra.get('uuid'),
+        )
+
+
+class NDChecklist(NDAnnotation, NDChecklistSubclass, VideoSupported):
+
+    @classmethod
+    def from_common(
+            cls, checklist: Checklist, schema_id: Cuid, extra: Dict[str, Any],
+            data: Union[VideoData, TextData, RasterData]) -> "NDChecklist":
+        return cls(answer=[
+            NDFeature(schema_id=answer.schema_id) for answer in checklist.answer
+        ],
+                   dataRow={'id': data.uid},
+                   schema_id=schema_id,
+                   uuid=extra.get('uuid'),
+                   frames=extra.get('frames'))
+
+
+class NDRadio(NDAnnotation, NDRadioSubclass, VideoSupported):
+
+    @classmethod
+    def from_common(cls, radio: Radio, schema_id: Cuid, extra: Dict[str, Any],
+                    data: Union[VideoData, TextData, RasterData]) -> "NDRadio":
+        return cls(answer=NDFeature(schema_id=radio.answer.schema_id),
+                   dataRow={'id': data.uid},
+                   schema_id=schema_id,
+                   uuid=extra.get('uuid'),
+                   frames=extra.get('frames'))
+
+
+class NDSubclassification:
+
+    @classmethod
+    def from_common(
+        cls, annotation: ClassificationAnnotation
+    ) -> Union[NDTextSubclass, NDChecklistSubclass, NDRadioSubclass]:
+        classify_obj = cls.lookup_subclassification(annotation)
+        if classify_obj is None:
+            raise TypeError(
+                f"Unable to convert object to MAL format. `{type(annotation.value)}`"
```

Comment:
I'm curious if this type (`NDSubclassification`) is to be used just for MAL or is a general type?

Reply:
Ah, nvm. I see the comment below.